azure/target/blobstorage/blobStorageFunction/index.js

```diff
@@ -0,0 +1,244 @@
+            }
+            mergedRecords[key] = value;
+        });
+        return mergedRecords;
+    } catch (e) {
+        return mergedRecords;
+    }
+};
+
+/**
+ * Processing logs by adding ingest_timestamp,log_type to log json object
+ * @param LogRecords
+ */
+const processLogs = (LogRecords, logSource) => {
+    const ingestionTime = Date.now();
+    if (typeof LogRecords === 'object') {
+        if (LogRecords !== undefined) {
+            fetchServiceName(LogRecords);
+            LogRecords.ingest_timestamp = ingestionTime;
+            LogRecords.log_type = 'azure_log';
+            LogRecords.logsource = logSource;
+        } else {
+            LogRecords.forEach((record) => {
+                fetchServiceName(record);
+                record.ingest_timestamp = ingestionTime;
+                record.log_type = 'azure_log';
+                record.logsource = logSource;
+            });
+        }
+    }
+};
+
+/**
+ * Fetch service name and provider from the resource ID string
+ * and adding those properties to event_provider,eventsource.
+ * @param LogRecords
+ */
+const fetchServiceName = (LogRecords) => {
+    if (!LogRecords.resourceId) {
+        error('Could not find resourceId for the log record');
+    }
+    var resourceInfo = LogRecords.resourceId.substring(1, LogRecords.resourceId.length);
+    var resourceArr = resourceInfo.split('/');
+    resourceArr.forEach(resource => {
+        if (resource.toUpperCase().startsWith("MICROSOFT.")) {
+            LogRecords.event_provider = "AZURE_" + resource.split('.')[1];
+        }
+    });
+    if(resourceArr[resourceArr.length - 2].toUpperCase()!=="PROVIDERS"){
+        LogRecords.eventsource = resourceArr[resourceArr.length - 2];
+    }
+};
+
+/**
+ *Processing and sending Log to vrli cloud for Event Hub.
+ * @param ServiceLogs
+ * @param collector
+ */
+const sendServiceLogsFromEventHub = (ServiceLogs, collector, trigger) => {
+    if (ServiceLogs[0].records) {
+        ServiceLogs.forEach((message) => {
+            message.records.forEach((record) => {
+                const data = collector.processLogsJsonFromEventHub(record);
+                collector.postDataToStream(data);
+            });
+        });
+    }
+};
+
+/**
+ * Processing and sending Log to vrli cloud for blob storage.
+ * @param serviceLogs
+ * @param collector
+ * @returns {Promise | Promise<any>}
+ */
+const sendServiceLogsFromBlobStorage = (serviceLogs, collector, trigger) => {
+    var logs;
+    if (typeof serviceLogs === 'string') {
+        logs = serviceLogs.trim().split('\n');
+    } else if (Buffer.isBuffer(serviceLogs)) {
+        logs = serviceLogs
+            .toString('utf8')
+            .trim()
+            .split('\n');
+    } else {
+        logs = JSON.stringify(serviceLogs)
+            .trim()
+            .split('\n');
+    }
+    logs.forEach(log => {
+        const data = collector.processLogsJsonFromBlobStorage(log);
+        return collector.postDataToStream(data);
```

Comment:
Here wont it return as soon as the first log gets processed?
i meant further logs will never be sent to LINT

Reply:
Have removed return statement and tested functionality, working as expected.